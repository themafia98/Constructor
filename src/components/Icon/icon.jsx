import React, {useState} from 'react';


function Icon(props){
    const [path] = useState(props.path);
    const [className] = useState(props.className ? props.className : null);
    const [draggable] = useState(props.draggable ? true : false);

    return (
    <img
        draggable = {draggable}
        onClick = {props.onClick ? props.onClick : null}
        className = {className}
        title = {props.title ? props.title : null}
        src = {path} alt = 'icon'>
    </img>
    )
}

export default Icon;