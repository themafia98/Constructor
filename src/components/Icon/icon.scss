[alt]{
    user-select: none;  
    // pointer-events: none;
}