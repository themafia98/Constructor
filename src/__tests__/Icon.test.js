import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../components/Icon/Icon';


const testEvent = event => {
    console.log('event work!');
}

const IconComponents = renderer.create(
    <Icon title = 'icon' className = "testIcon" path = '/img/plus.png' onClick = {testEvent} />
);

test('Icon.js test ', function () {

    let snaphotIcon = IconComponents.toJSON();
    expect(snaphotIcon).toMatchSnapshot();

    snaphotIcon.props.onClick({});

    snaphotIcon = IconComponents.toJSON();
    expect(snaphotIcon).toMatchSnapshot();
});