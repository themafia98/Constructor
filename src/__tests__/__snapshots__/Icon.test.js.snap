// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon.js test  1`] = `
<img
  alt="icon"
  className="testIcon"
  draggable={false}
  onClick={[Function]}
  src="/img/plus.png"
  title="icon"
/>
`;

exports[`Icon.js test  2`] = `
<img
  alt="icon"
  className="testIcon"
  draggable={false}
  onClick={[Function]}
  src="/img/plus.png"
  title="icon"
/>
`;
